cmd/kubectl-gadget/undeploy.go

```diff
@@ -192,32 +195,34 @@ again:
 	if undeployWait {
 		list, err = k8sClient.CoreV1().Namespaces().List(
 			context.TODO(), metav1.ListOptions{
-				FieldSelector: "metadata.name=" + utils.GadgetNamespace,
+				FieldSelector: "metadata.name=" + gadgetNamespace,
+				LabelSelector: "k8s-app=gadget",
```

Comment:
I don't see this label anymore on the gadget namespace. It needs to be removed here, otherwise it'll fail to remove the namespace.
btw, I think it's better not to label the namespace for now, that will make it impossible for users to remove an older instance of IG with a newer instance of kubectl-gadget.